### ISSUE TITLE
[Requirements] Additional parts about Annotations are needed

### ISSUE THREAD

**Issue by canatakan** [bounswe/bounswe2022group7#105]
In our [Requirements](https://github.com/bounswe/bounswe2022group7/wiki/Requirements-(Draft)), the only part about the Annotations is located under the "2. Non-Functional Requirements" part. That part only includes information about the standard that we will follow for Annotations.

However, we now need to update the Requirements according to the information in the [Annotation Research](https://github.com/bounswe/bounswe2022group7/wiki/Annotations-Research) such that:

* A new headline called "Annotations" shall be added under "1.1. User Requirements".
* All the content that can be annotated by the user shall be included under this headline.
* Both text and image annotation functions shall be described.

**Deadline: 12/04/2022 10:00**
**Reviewer: @canatakan**

**Comment by canatakan** [comment]
A quick note: While updating the "1.1. User Requirements", we should consider these points mentioned in the project description:

> Users should be able to annotate some context (text or image annotation) to clarify any points or to bookmark. These links may be to semantic resources which provide detailed information related to the resource.

> Users will be able to annotate content on the platform with text or images. The text may be a URI in which case it should serve as a hyperlink. These links may be to semantic resources which provide detailed information related to the resource.

**Comment by demet47** [comment]
### 1.1.13 Annotations
* 1.1.13.1: Registered users shall be able to annotate text pieces with corresponding links.
    * 1.1.13.1.1: Registered users shall be able to annotate text pieces in discussion posts.
    * 1.1.13.1.2: Registered users shall be able to annotate text pieces in comment sections.
    * 1.1.13.1.3: Artists shall be able to annotate text pieces in their own exhibition descriptions and online gallery descriptions.
    * 1.1.13.1.4: Artists shall be able to annotate text pieces in their own art item descriptions.
* 1.1.13.2: Registered users shall be able to annotate various content by adding custom labels.
    * 1.1.13.2.1: Registered users shall be able to label discussion posts.
    * 1.1.13.2.2: Artists shall be able to label their own exhibition and online galleries.
    * 1.1.13.2.3: Artists shall be able to label their own art items.
* 1.1.13.3: Registered users shall be able to bookmark discussion posts, art items, exhibitions and online galleries.

**Comment by demet47** [comment]
> We tought of cases where we would use annotations. The ones we decided upon are:
> 
> * Adding hyperlinked texts to comments, discussion posts and online gallery-physical exhibition descriptions.
> * Labeling discussion posts, online gallery-physical exhibitions and art items
> * Bookmarking discussion posts, art items, online gallery-physical exhibitions

Correspond

**Comment by demet47** [comment]
> We tought of cases where we would use annotations. The ones we decided upon are:
> 
> * Adding hyperlinked texts to comments, discussion posts and online gallery-physical exhibition descriptions.
> * Labeling discussion posts, online gallery-physical exhibitions and art items
> * Bookmarking discussion posts, art items, online gallery-physical exhibitions

We wrote  corresponding User Requirements parts as follows with @canatakan:

### 1.1.13 Annotations
* 1.1.13.1: Registered users shall be able to annotate text pieces with corresponding links.
    * 1.1.13.1.1: Registered users shall be able to annotate text pieces in discussion posts.
    * 1.1.13.1.2: Registered users shall be able to annotate text pieces in comment sections.
    * 1.1.13.1.3: Artists shall be able to annotate text pieces in their own exhibition descriptions and online gallery descriptions.
    * 1.1.13.1.4: Artists shall be able to annotate text pieces in their own art item descriptions.
* 1.1.13.2: Registered users shall be able to annotate various content by adding custom labels.
    * 1.1.13.2.1: Registered users shall be able to label discussion posts.
    * 1.1.13.2.2: Artists shall be able to label their own exhibition and online galleries.
    * 1.1.13.2.3: Artists shall be able to label their own art items.
* 1.1.13.3: Registered users shall be able to bookmark discussion posts, art items, exhibitions and online galleries.

**Comment by demet47** [comment]
> Reference i

I wrote the corresponding System Requirements parts as follows:

### 1.2.9 Annotation
* 1.2.9.1: System shall allow registered users to create hyperlink for a text piece the user writes. [1.1.13.1]
    * 1.2.9.1.1: System shall allow registered users to create hyperlink for a text piece the user writes in their discussion post.[1.1.13.1.1]
    * 1.2.9.1.2: System shall allow registered users to create hyperlink for a text piece the user writes in comment sections for exhibitions and art items.[1.1.13.1.2]
    * 1.2.9.1.3: System shall allow artists to create hyperlink for a text piece the artist writes in their own exhibitions' descriptions and online galleries' descriptions.[1.1.13.1.3]
    * 1.2.9.1.4: System shall allow artists to create hyperlink for a text piece the artist writes in their own art items' descriptions.[1.1.13.1.4]
* 1.2.9.2: System shall allow registered users to annotate content with customized textual data.[1.1.13.2]
    * 1.2.9.2.1: System shall allow registered user to annotate discussion posts they create with corresponding customized textual data.[1.1.13.2.1]
    * 1.2.9.2.2: System shall allow registered user to annotate the exhibitions and online galleries they create with corresponding customized textual data.[1.1.13.2.2]
    * 1.2.9.2.3: System shall allow registered user to annotate their own art items with corresponding customized textual data.[1.1.13.2.3]
* 1.2.9.1.3: System shall allow annotating discussion posts, art items, exhibitions and online galleries target URL being a link implying a page with collection of bookmarked items and an empty body (see [Annotations Research *Bookmarking Content* part](https://github.com/bounswe/bounswe2022group7/wiki/Annotations-Research#bookmarking-content)). [1.1.13.3]
Review request: @canatakan

**Comment by demet47** [comment]
For search feature, we designed a structure for annotation where when we search a label the items annotated with that label will be listed. @oguzpancuk (@CahidArda being the reviewer) are assigned for thorough completion of search feature, so this should be taken into consideration, beware.

**Comment by canatakan** [comment]
The review is now completed. Notes:

* I made minor improvements in the language.
* I do not think that we should provide any information as to how the annotating will be implemented in Requirements. Hence, removed the reference to the [Annotations Research](https://github.com/bounswe/bounswe2022group7/wiki/Annotations-Research).
* Renamed "1.2.9 Annotation" as "1.2.9 Annotations" and updated the Table of Contents.

Apart from those, User Requirements are exactly as we discussed in the meeting and the System Requirements that you have written also reflect the necessary functionalities.

Thank you for your cooperation, and meticulous work. Then, the task is done, hence closing the issue.
